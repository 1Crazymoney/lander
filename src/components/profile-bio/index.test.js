import React from 'react';
import ProfileBio from './index';
import renderer from 'react-test-renderer';

it('not in edit mode and bio is empty. should render null', () => {
  const props = {
    toggleUiProp: () => {
    }
  };

  const component = <ProfileBio {...props} />;

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('in edit mode. bio is empty. should show placeholder', () => {
  const props = {
    editMode: true
  };

  const component = <ProfileBio {...props} />;

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('bio is not empty in edit mode. should render bio', () => {
  const props = {
    editMode: true,
    bio: 'Lorem ipsum dolor sit amet'
  };

  const component = <ProfileBio {...props} />;

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


it('bio is not empty in not edit mode. should render bio (without onclick event)', () => {
  const props = {
    bio: 'Lorem ipsum dolor sit amet'
  };

  const component = <ProfileBio {...props} />;

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
