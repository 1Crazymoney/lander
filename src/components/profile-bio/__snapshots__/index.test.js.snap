// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bio is not empty in edit mode. should render bio 1`] = `
<div
  className="profile-bio edit-mode"
  onClick={[Function]}
>
  <p>
    Lorem ipsum dolor sit amet
  </p>
</div>
`;

exports[`bio is not empty in not edit mode. should render bio (without onclick event) 1`] = `
<div
  className="profile-bio"
>
  <p>
    Lorem ipsum dolor sit amet
  </p>
</div>
`;

exports[`in edit mode. bio is empty. should show placeholder 1`] = `
<div
  className="profile-bio not-set edit-mode"
  onClick={[Function]}
>
  <span>
    Add longer description to your profile
  </span>
</div>
`;

exports[`not in edit mode and bio is empty. should render null 1`] = `null`;
